packages/opentelemetry-api/src/context/propagation/getter.ts

```diff
@@ -14,6 +14,11 @@
  * limitations under the License.
  */
 
-export type Carrier = {
-  [key: string]: unknown;
-};
+export type GetterFunction<Carrier = any> = (
+  carrier: Carrier,
+  key: string
+) => unknown;
+
+export function defaultGetter(carrier: any, key: string): unknown {
+  return carrier[key];
```

Comment:
How do we handle case insensitive for a HTTP request object (per specs: if the getter is intended to work with a HTTP request object, the getter MUST be case insensitive)?

Reply:
This is just meant to be the default behavior if you don't pass in a getter function. It just prevents the propagators from having to do `if (getter) { } else {}` every time. They can just use `getter` without thinking about it. I'm not sure how this can be made case insensitive without giving up quite a bit of efficiency and looping over all entries.